packages/cli/test/demo/names-in-transit.js

```diff
@@ -0,0 +1,31 @@
+/** @import {Context, TestRoutine} from '../types' */
+
+/** @type {TestRoutine} */
+export const section = async (execa, testLine) => {
+  // In this example, we send alice our "doubler" but let it appear...
+  await testLine(
+    execa`endo send alice ${'Please enjoy this @counter:doubler.'}`,
+  );
+  await testLine(execa`endo inbox --as alice-agent`, {
+    stdout: /^1\. "HOST" sent "Please enjoy this @counter\."/,
+  });
+  await testLine(execa`endo adopt --as alice-agent 1 counter --name redoubler`);
+  await testLine(execa`endo list --as alice-agent`, {
```

Comment:
I imagine this is the first thing that fails in this test. There's an asymmetry in the API here: rather than `endo list --as <agent>`, you simply do `endo list <agent>`. Does that change cause these test cases to succeed?

Reply:
I think your suggestion does what you think it does, but I want to limit the scope of this PR to correctly implementing a test of whether the expectations set in the demo's README.md match the observations from running the demo's commands. A future PR could make these tests pass either by changing the behavior of endo/cli or by altering the demo's expectations. In the former case it will be satisfying to associate the fixing change to the removal of `failing` in these tests. In the latter case it will be nice to have an example PR conjoining edits to the README.md with edits to these tests.

Reply:
I see, although in this case it's the README that's wrong; `--as` was omitted from `list` in a deliberate design decision. I noticed the discrepancy myself a while back and even put up a PR for it (#2228), but we decided against making the change at the time. I don't think the situation is likely to change unless we complete a more thorough audit of each command for `--as` support, and until then I think it makes the most sense to update the readme and these tests.

Reply:
I don't think we actually resolved this point, but I'll let it go in exchange for you resurrecting #2228 😉